src/radiation/moments_source.cpp

```diff
@@ -351,12 +353,26 @@ TaskStatus MomentFluidSourceImpl(T *rc, Real dt, bool update_fluid) {
             // Rootfind over fluid temperature in fluid rest frame
             root_find::RootFind root_find(src_rootfind_maxiter);
             InteractionTResidual res(eos, opacities, rho, ug, Ye, J0, num_species,
-                                     species_d, dtau);
+                                     species_d, dtau, alpha_max);
             root_find::RootFindStatus status;
-            Real T1 = root_find.itp(res, 1.e-5 * v(iblock, pT, k, j, i),
-                                    1.e5 * v(iblock, pT, k, j, i),
-                                    src_rootfind_tol * v(iblock, pT, k, j, i),
-                                    v(iblock, pT, k, j, i), &status);
+            Real T1 = root_find.regula_falsi(res, 1.e-2 * v(iblock, pT, k, j, i),
+                                             1.e2 * v(iblock, pT, k, j, i),
+                                             src_rootfind_tol * v(iblock, pT, k, j, i),
+                                             v(iblock, pT, k, j, i), &status);
+            Real rebracketing_fac = 10.;
+            int n_rebracketing_tries = 0;
+            constexpr int MAX_REBRACKETING_TRIES = 4;
+            while (status == root_find::RootFindStatus::failure &&
+                   n_rebracketing_tries < MAX_REBRACKETING_TRIES) {
+              T1 = root_find.regula_falsi(
+                  res, 1.e-1 / rebracketing_fac * v(iblock, pT, k, j, i),
+                  1.e1 * rebracketing_fac * v(iblock, pT, k, j, i),
```

Comment:
The original bracket window of 10 decades could cause precision issues in some cases, especially with the `itp` rootfinder.